package/src/components/Chat/Chat.tsx

```diff
@@ -198,15 +199,29 @@ const ChatWithContext = <
   const setActiveChannel = (newChannel?: Channel<StreamChatGenerics>) => setChannel(newChannel);
 
   useEffect(() => {
-    if (userID && enableOfflineSupport) {
+    if (!(userID && enableOfflineSupport)) return;
+
+    const initializeDatabase = () => {
       // This acts as a lock for some very rare occurrences of concurrency
       // issues we've encountered before with the QuickSqliteClient being
       // uninitialized before it's being invoked.
-      setInitialisedDatabaseConfig({ initialised: false, userID });
-      SqliteClient.initializeDatabase();
-      setInitialisedDatabaseConfig({ initialised: true, userID });
-      DBSyncManager.init(client as unknown as StreamChat);
-    }
+      SqliteClient.initializeDatabase()
```

Comment:
```suggestion
      setInitialisedDatabaseConfig({ initialised: false, userID });
      SqliteClient.initializeDatabase()
```

Please return the safeguard here unless it's causing issues. The reason being: If the DB is getting initialized again for one reason or another (for example, the `userID` changed) we need this to safeguard any potential DB queries or pending tasks trying to be executed in the background.